lib/src/client.dart

```diff
@@ -114,7 +113,7 @@ class ClientImpl implements Client {
   ClientConfig get config => _config;
 
   String _token = '';
-  List<int>? _data;
+  List<int>? get _data => _config.data;
```

Comment:
@InfernalRabbit thanks for the example, much more clear now! Looking at the example you provided – should this getter be called `data` instead of `_data`?

But ideally, can this be made in a slightly different way - we also have this functionality in `centrifuge-js`:

https://github.com/centrifugal/centrifuge-js/blob/c825ddcc5056a8660c694f6f79fee7c73d2907a4/src/types.ts#L103

So that config have `data` and natively supports optional `getData` helper to pass a dynamic data to the Client. If this helper is set - it's always called by SDK instead of using `data` field. Will it work for your use case? If yes - it's preferable way, as it will allow keeping SDKs similar (even if a bit less idiomatic for the language). It will be possible to add this to the common client SDK spec later.

Reply:
@FZambia Checked out JS lib, and that solution will work. I can make changes to be similar

_data should be named _data because this getter is internal only and there is a logic confusion if _data will be public. You can get every configuration field from config getter

Reply:
Thank you, yep - let's make it similar to centrifuge-js then